cargo-dylint/tests/ci.rs

```diff
@@ -713,6 +713,65 @@ fn update() {
     });
 }
 
+#[test]
+fn lint() {
+    let mut restriction_libs = Vec::new();
+    for entry in read_dir("examples/restriction").unwrap() {
+        let entry = entry.unwrap();
+        if entry.path().is_dir() {
+            let lib_name = entry.file_name().into_string().unwrap();
+            // Exclude overscoped_allow as in the script
+            if lib_name != "overscoped_allow" && lib_name != ".cargo" {
+                restriction_libs.push(format!("--lib {lib_name}"));
+            }
+        }
+    }
+    let restrictions_as_flags = restriction_libs.join(" ");
+
+    let mut dirs_to_lint: Vec<PathBuf> = [
+        ".",
+        "driver",
+        "utils/linting",
+        "examples/general",
+        "examples/supplementary",
+        "examples/restriction",
+        "examples/testing/clippy",
+    ]
+    .into_iter()
+    .map(PathBuf::from)
+    .collect::<Vec<_>>();
+
+    for entry in read_dir("examples/experimental").unwrap() {
+        let entry = entry.unwrap();
+        let path = entry.path(); // path is already a PathBuf
+        if path.is_dir() && path.file_name().unwrap() != ".cargo" {
+            dirs_to_lint.push(path);
+        }
+    }
+
+    let base_flags =
```

Comment:
Could you please move this declaration to between the declarations of `restrictions_as_flags` and `dirs_to_lint`?